gammapy/background/off_data_background_maker.py

```diff
@@ -203,3 +211,59 @@ def save_models(self, modeltype):
         """
         for ngroup in range(self.ntot_group):
             self.save_model(modeltype, ngroup)
+
+    def background_links(self, data_store, modeltype, out_dir_background_model=None, outfile=None,
+                            filename_obs_group_table=None):
+        """Creates a link for each observation of the obs_table to the corresponding background model
+
+        Parameters
+        ----------
+        data_store : `~gammapy.data.DataStore`
+            Create a `~gammapy.data.DataStore` for the runs for which ones we want to compute a background model
+        modeltype : {'3D', '2D'}
+            Type of the background modelisation
+        out_dir_background_model:  str
+            directory where are located the backgrounds models for each band in zenith and efficiency
+        outfile: str
+            name of the new HDU-index file that is an `~astropy.table.Table` with tha row for the background model
+        filename_obs_group_table : str
+            name of the file containing the `~astropy.table.Table` with the group infos
+
+
+        """
+        obs_table = data_store.obs_table
+        if not filename_obs_group_table:
+            filename_obs_group_table = self.group_table_filename
+        if not out_dir_background_model:
+            out_dir_background_model = data_store.hdu_table.meta["BASE_DIR"]
+        if not outfile:
+            outfile = Path("hdu-index2.fits.gz")
+        table_group = Table.read(filename_obs_group_table, format='ascii.ecsv')
+        axes = ObservationGroups.table_to_axes(table_group)
+        groups = ObservationGroups(axes)
+        obs_table = ObservationTable(obs_table)
+        obs_table = groups.apply(obs_table)
+        index_table_bkg = Table(names=("OBS_ID", "HDU_TYPE", "FILE_DIR", "FILE_NAME", "HDU_NAME", "HDU_CLASS"),
```

Comment:
You shouldn't have to manually set the dtype.
Especially with string data it's easy to get corrupt data if you set it too short and then it's truncated.

I usually use this pattern to construct tables and let the Table constructor choose all dtypes:
```python
In [1]: %paste
# Somehow create table data as a list of dicts
data = []
for i in range(3):
    row = {}
    row['a_number'] = i
    row['a_string'] = 'hi ' * (i + 1)
    data.append(row)

# Create a table object
from astropy.table import Table
table = Table(data)
print(data)
print(table)

## -- End pasted text --
[{'a_number': 0, 'a_string': 'hi '}, {'a_number': 1, 'a_string': 'hi hi '}, {'a_number': 2, 'a_string': 'hi hi hi '}]
a_number  a_string
-------- ---------
       0       hi 
       1    hi hi 
       2 hi hi hi 
```